src/scenex/adaptors/_pygfx/_points.py

```diff
@@ -28,52 +29,72 @@ class Points(Node):
+        )
+        self._pygfx_node = pygfx.Points(None, self._material)
+        self._snx_set_coords(points.coords)
+
+    def _create_geometry(self, coords: npt.NDArray | None) -> pygfx.Geometry:
         # TODO: unclear whether get_view() is better here...
-        coords = np.asarray(points.coords)
+        coords = np.asarray(coords)
         n_coords = len(coords)
 
         # ensure (N, 3)
         if coords.shape[1] == 2:
             coords = np.column_stack((coords, np.zeros(coords.shape[0])))
 
         geo_kwargs = {}
-        if points.face_color is not None:
-            colors = np.tile(np.asarray(points.face_color), (n_coords, 1))
+        if self._model.face_color is not None:
+            colors = np.tile(np.asarray(self._model.face_color), (n_coords, 1))
             geo_kwargs["colors"] = colors.astype(np.float32)
 
-        # TODO: not sure whether/how pygfx implements all the other properties
-
-        self._geometry = pygfx.Geometry(
+        return pygfx.Geometry(
             positions=coords.astype(np.float32),
-            sizes=np.full(n_coords, points.size, dtype=np.float32),
+            sizes=np.full(n_coords, self._model.size, dtype=np.float32),
             **geo_kwargs,
         )
-        self._material = pygfx.PointsMaterial(
-            size=points.size,  # pyright: ignore[reportArgumentType]
-            size_space=SPACE_MAP[points.scaling],
-            aa=points.antialias > 0,
-            opacity=points.opacity,
-            color_mode="vertex",
-            size_mode="vertex",
-        )
-        self._pygfx_node = pygfx.Points(self._geometry, self._material)
 
-    def _snx_set_coords(self, coords: npt.NDArray) -> None: ...
+    def _snx_set_coords(self, coords: npt.NDArray | None) -> None:
+        self._pygfx_node.geometry = self._create_geometry(coords)
+
+    def _snx_set_size(self, size: float) -> None:
+        n_coords = len(self._model.coords)
+        sizes = np.full(n_coords, self._model.size, dtype=np.float32)
+        self._pygfx_node.geometry.sizes = pygfx.Buffer(sizes)  # pyright: ignore[reportOptionalMemberAccess]
 
-    def _snx_set_size(self, size: float) -> None: ...
+    def _color_buffer(self, color: Color | None) -> pygfx.Buffer:
+        if color is None:
+            color = cmap.Color("transparent")
+        n_coords = len(self._model.coords)
+        colors = np.tile(np.asarray(color), (n_coords, 1))
+        return pygfx.Buffer(colors.astype(np.float32))
 
-    def _snx_set_face_color(self, face_color: Color) -> None: ...
+    def _snx_set_face_color(self, face_color: Color | None) -> None:
+        self._pygfx_node.geometry.colors = self._color_buffer(face_color)  # pyright: ignore[reportOptionalMemberAccess]
 
-    def _snx_set_edge_color(self, edge_color: Color) -> None: ...
+    def _snx_set_edge_color(self, edge_color: Color | None) -> None:
+        self._pygfx_node.geometry.edge_color = self._color_buffer(edge_color)  # pyright: ignore[reportOptionalMemberAccess]
```

Comment:
Does this work for you? I'm not seeing an edge color...

![image](https://github.com/user-attachments/assets/c0e8cce8-ad8d-42fe-91e8-6c411457f15d)

Reply:
Notably, I **can** change the face color...

Reply:
Indeed, there are a number of not-quite-right things that this tool makes much easier to find 😀 

We need to take a closer look at the points adapter